### DISCUSSION TITLE
Third-party module import

### DISCUSSION THREAD

**Discussion by Minecodes** [pyscript/pyscript#1195] [Proposals]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other feature requests and couldn't find a duplicate (including also the ``type-feature`` tag)
- [X] I confirmed that it's not related to another project area (see the above section)

### What is the idea?

It should be a function, that imports modules from platforms like pip (PyPi).
In other words the functionality of `pip install`.

### Why is this needed

This would be good for users that use libraries that are not included.

### What should happen?

A user would use the tag `<py-import>` to import third-party modules and pyscript would download it into the cache or into the WASM file system.

For example:
```html
<py-import>
mintotp
requests
</py-import>

<py-script>
print(mintotp.totp('ZYTYYE5FOAGW5ML7LRWUL4WTZLNJAMZS'))

r = requests.get('https://httpbin.org/ip')
print('HTTP Code: ' + r.status_code)
print('JSON Output: ' + r.json())
</py-script>
```

**Comment by JeffersGlass** [comment]
Unless I'm misunderstanding your intent, this functionality is already present in [the <py-env> tag](https://github.com/pyscript/pyscript/blob/479348eec9739eed3b7c4bdf98cf1dedbc303be7/docs/tutorials/getting-started.md#the-py-env-tag), which loads the listed packages from the [pyodide built-in packages](https://pyodide.org/en/stable/usage/packages-in-pyodide.html) if available, or tries to load them from PyPI if not. Typically, the latter will only work with pure Python packages. See [Pyodide's documentation on loading packages](https://pyodide.org/en/stable/usage/loading-packages.html#loading-packages) for more info.

**Comment by ntoll** [comment]
Hmm... I think @Minecodes might also be asking for `pip` syntax (as used in, for instance, `requirements.txt`). The existing `py-env` tag uses YAML (being relatively new here, I'm not sure why). Perhaps this could be an enhancement so the `py-env` syntax is the same as `pip` understands? Familiarity for Python developers would be a good side effect of this change.

**Comment by Minecodes** [comment]
> Hmm... I think @Minecodes might also be asking for `pip` syntax (as used in, for instance, `requirements.txt`). The existing `py-env` tag uses YAML (being relatively new here, I'm not sure why). Perhaps this could be an enhancement so the `py-env` syntax is the same as `pip` understands? Familiarity for Python developers would be a good side effect of this change.

yes, that's what I mean and I didn't found it in the docs.

**Comment by marimeireles** [comment]
Also this one @madhur-tandon.
Thanks!!!
